sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -210,7 +210,7 @@ public final Decision shouldSample(
       // while allowing for a (very) small chance of *not* sampling if the id == Long.MAX_VALUE.
       // This is considered a reasonable tradeoff for the simplicity/performance requirements (this
       // code is executed in-line for every Span creation).
-      return Math.abs(traceId.getLowerLong()) < getIdUpperBound()
+      return Math.abs(traceId.getTraceRandomPart()) < getIdUpperBound()
```

Comment:
We need to make sure every library applies the same logic and extract correctly the random part. This is important because if 2 libraries are configured with the same probability we need both to sample or not sample a given trace id.

Reply:
I was looking at other languages how they implemented this, and it's all over the place. Looking at the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#built-in-samplers), the only indication is to use part of the trace for the decision.